test/fixtures/files/pay/invoice.upcoming.json

```diff
@@ -0,0 +1,120 @@
+{
```

Comment:
I love the idea of using the fixture but I'm not too happy about 120 lines of new code.

Can we trim this down to the bare minimum needed to get the test passing?

Reply:
We probably can, but is it a better solution? We're providing the Stripe's webhook data to Pay. If the data is not complete (as you're suggesting) and Pay changes something, for example starts processing another line from the data, the test could fail - and not because our code isn't working.

In my opinion, it's better to just copy-paste the same file Pay uses for testing the webhook, even if it's 120-line fixture. But we pretty much don't care about the content - except two lines specified in the test. And if Stripe changes something in the webhook data, we'll just need to copy-paste the whole file.

Again, if you really want to trim this down, let me know. 🙂

Reply:
This is probably the smallest footprint:
```ruby
Stripe::Event.construct_from(
  data: {
    object: {
      subscription: subscription.processor_id,
      next_payment_attempt: renewal_timestamp,
      lines: {
        data: [
          {
            price: {
              type: "recurring",
              recurring: {interval: "month"}
            }
          }
        ]
      }
    }
  }
)
```

Reply:
Good point - lets keep the entire file then!